bin/commands/support/index.ts

```diff
@@ -0,0 +1,237 @@
+
+import * as std from 'cm_std';
+import * as zos from 'zos';
+import * as xplatform from 'xplatform'
+
+import * as common from '../../libs/common';
+import * as component from '../../libs/component';
+import * as config from '../../libs/config';
+import * as fs from '../../libs/fs';
+import * as java from '../../libs/java';
+import * as node from '../../libs/node'
+import * as shell from '../../libs/shell';
+import * as zoslib from '../../libs/zos';
+import * as zosfs from '../../libs/zos-fs';
+import * as zosmf from '../../libs/zosmf';
+
+import * as verifyFingerprints from './verify-fingerprints/index';
+
+function zssCheck(zssBinary: string): string {
+  if (fs.fileExists(zssBinary)) {
+      return `${zssBinary} = ${component.hasPCBit(zssBinary)}`
+  } else {
+      common.printError(`Error ZWEL0150E: Failed to find file "${zssBinary}". Zowe runtimeDirectory is invalid.`);
+      return `Missing file: ${zssBinary}`
+  }
+}
+
+export function execute(): void {
+
+  common.printLevel0Message('Collect information for Zowe support');
+  const isoDate = new Date().toISOString().replace(/T|t|:/g,'-').substring(0,19);
+
+  let targetDirectory = std.getenv('ZWE_CLI_PARAMETER_TARGET_DIR');
+  if (targetDirectory) {
+    targetDirectory = fs.convertToAbsolutePath(targetDirectory);
+  } else {
+    targetDirectory = fs.getTmpDir();
+  }
+
+  if (!fs.directoryExists(targetDirectory, true)) {
+    common.printErrorAndExit(`Error ZWELxxxx: "${targetDirectory}" is not a valid directory.`, undefined, 999);
+  }
+
+  const tmpFilePrefix = 'zwe-support';
+  const tmpPax = `${targetDirectory}/${tmpFilePrefix}.${isoDate}.pax`;
+  const tmpDir = fs.createTmpFile(tmpFilePrefix, targetDirectory);
+
+  common.requireZoweYaml();
+  const ZOWE_CONFIG=config.getZoweConfig();
+
+  common.printMessage(`Started at ${isoDate}`);
+  fs.mkdirp(tmpDir, 0o700);
+  common.printDebug(`Temporary directory created: ${tmpDir}`);
+
+  common.printLevel1Message('Collecting various environment information');
+  const environmentFile =`${tmpDir}/environment.json`;
+  let environment = {};
+
+  environment["zos-version"] = zoslib.formatZosVersion();
+
+  const enabledComponents = component.getEnabledComponents();
+  if (enabledComponents.includes('app-server')) {
+    node.requireNode();
+  }
+  const nodeHome = std.getenv('NODE_HOME');
+  if (nodeHome) {
+    const nodeVersion = shell.execOutSync('sh', '-c', '${NODE_HOME}/bin/node -v 2>&1 | head -n 1');
+    if (nodeVersion.rc == 0 && nodeVersion.out) {
+      environment["node"] = `${nodeVersion.out}`;
+      const discovery = ZOWE_CONFIG.components?.discovery?.enabled;
+      const zosmfHost = ZOWE_CONFIG.zOSMF?.host;
+      const zosmfPort = ZOWE_CONFIG.zOSMF?.port;
+      if (discovery && zosmfHost && zosmfPort) {
+        environment["zosmf_check"] = `'https://${zosmfHost}:${zosmfPort}/zosmf/info' => ${zosmf.validateZosmfHostAndPort(zosmfHost, zosmfPort)}`;
+      }
+    }
+  } else {
+    environment["node"] = `not found`;
```

Comment:
should we add environment["zosmf_check"] = `<url> => not checked` ? Capturing the URL could still be useful when node is not available. Second question: is it useful to capture this when discovery is disabled?

Reply:
Following code was made based on [`internal start prepare`](https://github.com/zowe/zowe-install-packaging/blob/e394856fd82e8316bde4bdaebdbb344e40721555/bin/commands/internal/start/prepare/index.ts#L168-L180):
* If `discovery`, check `zOSMF` -> when we really need `zOSMF`
* Other possible cases not supported
* URL can be always recreated from `zowe.yaml` config

I am OK to change this to try to do a check in more cases.